src/smc-webapp/editor.coffee

```diff
@@ -574,9 +572,10 @@ templates = $("#salvus-editor-templates")
 #
 
 class FileEditor extends EventEmitter
-    constructor: (@project_id, @filename, content, opts) ->
+    # ATTN it is crucial to call this constructor in subclasses via super(@project_id, @filename)
```

Comment:
I would be more comfortable if this was enforced some way.  E.g.,  in the code that does `new ThingThatDervicesFromFileEditor`, have it check that constructor was called, or at least the reason for it being called holds, e.g., that @ext is defined.   That would be 2 lines of code and would make it impossible for anybody to ever have an editor work at all that doesn't call super properly.